Coffee-Shop-Simulation/Machine.java

```diff
@@ -0,0 +1,86 @@
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * A Machine is used to make a particular Food.  Each Machine makes
+ * just one kind of Food.  Each machine has a machineCap: it can make
+ * that many food items in parallel; if the machine is asked to
+ * produce a food item beyond its machineCap, the requester blocks.
+ * Each food item takes at least item.cookTimeMS milliseconds to
+ * produce.
+ */
+public class Machine {
+    public final String machineName;
+    public final Food machineFoodType;
+
+    //YOUR CODE GOES HERE...
+    int machineCap;
+    List<Food> foodList;
+
+    /**
+     * The constructor takes at least the name of the machine,
+     * the Food item it makes, and its machineCap.  You may extend
+     * it with other arguments, if you wish.  Notice that the
+     * constructor currently does nothing with the machineCap; you
+     * must add code to make use of this field (and do whatever
+     * initialization etc. you need).
+     */
+    public Machine(String machineName, Food food, int capacity) {
+        //YOUR CODE GOES HERE...
+        this.machineFoodType = food;
+        this.machineCap = capacity;
+        this.foodList = new ArrayList<Food>();
+        this.machineName = machineName;
+    }
+
+
+    /**
+     * This method is called by a Cook in order to make the Machine's
+     * food item.  You can extend this method however you like, e.g.,
+     * you can have it take extra parameters or return something other
+     * than Object.  It should block if the machine is currently at full
+     * machineCap.  If not, the method should return, so the Cook making
+     * the call can proceed.  You will need to implement some means to
+     * notify the calling Cook when the food item is finished.
+     */
+    public void makeFood(Cook name, int orderNum) throws InterruptedException {
+        //YOUR CODE GOES HERE...
+        foodList.add(machineFoodType);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.



**Problem**  
 This line may not be thread-safe: **48**. At this line the following field is not protected by synchronization: `foodList`. Your code typically protects this field with synchronization on object: `foodList`.

**Fix**  
 Consider protecting the above field with synchronization on object: `foodList`. For example, like at lines: **74**, **75**, and **73**.

**More info**  
 The following lines are where the above field is already protected by synchronization on the above object: **73**, **74**, and **75**.